ow_plexil/src/plexil-adapter/OwAdapter.cpp

```diff
@@ -283,40 +283,32 @@ static void discard (Command* cmd, AdapterExecInterface* intf)
   acknowledge_command_sent(*cr);
 }
 
-static void tilt_antenna (Command* cmd, AdapterExecInterface* intf)
+static bool check_closed_range (const char* name, double val,
+                                double min, double max, double tolerance)
```

Comment:
Can you explain the need for a tolerance?

Reply:
For validating pan/tilt input, a tiny tolerance is needed to deal with floating point representation, e.g. 1 getting represented as 0.9999  or 1.0001.

As to whether a tolerance belongs in a function named "check_closed_range" is a separate issue, but this copies exactly what's already in ow_lander/scripts/utils.py.

Open to more discussion.

Reply:
I did a little more digging into this issue. Python and PLEXIL both store floats as double-precision (64bit), but we have defined all of our ROS Actions to use Float32. The problem arises the same way whether PLEXIL or the action client python script (command line) call the action. The float starts as a double, is squeezed into a Float32 and loses precision, and then regains precision when it is converted to a double again on the action's server side (also a Python script). I tested this by printing the full float representation in both the client and server. For certain floats, the server-side version was altered. Then I redefined **AntennaPanTilt.action** to use only Float64. The altered floats on the server-side went away entirely when Float64 was used.
_Note_ I performed this test using only the action client and not PLEXIL, so there could be some other strangeness going on there. 

I think this function is an acceptable work-around to a problem that will be fixed when we complete command unification, which requires that all our action files use Float64. I think it's fine to keep the tolerance for now, but lets make a note somewhere to remove it here and in utils.py when command unification efforts are complete.

Reply:
> e.g. 1 getting represented as 0.9999 or 1.0001.

I could not actually replicate this exact example in my tests. Even when **AntennaPanTilt.action** used Float32, a "1.0" client-side would convert to a "1.0" server-side. However, "1.1" client-side  would convert to a "1.100000023841858" server-side.

Reply:
https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-1085

Reply:
Thanks for looking into this.  I'll add a comment in both functions (C++ and Python) about this issue, and agree it should in principle go away after float64 is used in the action.

My example using 1 was made up!  I guess it was a bad example.